Content.Server/GameObjects/EntitySystems/AI/Pathfinding/Queues/PathfindingPriorityQueue.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.EntitySystems.AI.Pathfinding.Queues
+{
+    public interface IPathfindingPriorityQueue<T>
+    {
+        bool Contains(T key);
+        int Count { get; }
+        void Enqueue(T item, float priority);
+        T Dequeue();
+    }
+
+    // Pretty crappy, replace with something better
+    public class PathfindingPriorityQueue<T> : IPathfindingPriorityQueue<T>
+    {
+        private readonly List<Tuple<T, float>> _elements = new List<Tuple<T, float>>();
```

Comment:
For future reference, .NET's `Tuple` (not to be confused with `ValueTuple` is kinda bad because it's a reference type, making it quite expensive. Prefer to use `ValueTuple` instead.

(I know there's no other ones in this PR but still figured it was worth mentioning)